tests/core/test_config.py

```diff
@@ -593,6 +593,20 @@ async def test_raw_with_partial_primary_keys(config):
     assert await config.custom("CUSTOM", "primary_key").identifier() is False
 
 
+@pytest.mark.asyncio
+async def test_cast_subclass_default(config):
+    from collections import Counter
+
+    config.register_global(foo=Counter({}))
```

Comment:
I think we should assert that the value stored in `defaults` is not a Counter instance as well.
```suggestion
    config.register_global(foo=Counter({}))
    assert type(config.defaults["GLOBAL"]["foo"]) is dict
    assert config.defaults["GLOBAL"]["foo"] == {}
```